src/stcal/ramp_fitting/ols_fit.py

```diff
@@ -113,246 +109,36 @@ def ols_ramp_fit_multi(
-                if resultslice[11] is not None:  # Optional results exist
-                    opt_model.slope[:, :, start_row:total_rows, :] = resultslice[11]
-                    opt_model.sigslope[:, :, start_row:total_rows, :] = resultslice[12]
-                    opt_model.var_poisson[:, :, start_row:total_rows, :] = resultslice[13]
-                    opt_model.var_rnoise[:, :, start_row:total_rows, :] = resultslice[14]
-                    opt_model.yint[:, :, start_row:total_rows, :] = resultslice[15]
-                    opt_model.sigyint[:, :, start_row:total_rows, :] = resultslice[16]
-                    opt_model.pedestal[:, start_row:total_rows, :] = resultslice[17]
-                    opt_model.weights[:, :, start_row:total_rows, :] = resultslice[18]
-                    opt_model.crmag[:, :, start_row:total_rows, :] = resultslice[19]
-            else:  # all but last slice
-                stop_row = (k + 1) * rows_per_slice
-                out_model.data[start_row: stop_row, :] = resultslice[0]
-                out_model.dq[start_row: stop_row, :] = resultslice[1]
-                out_model.var_poisson[start_row: stop_row, :] = resultslice[2]
-                out_model.var_rnoise[start_row: stop_row, :] = resultslice[3]
-                out_model.err[start_row: stop_row, :] = resultslice[4]
-                if resultslice[5] is not None:  # Multiple integration results exist
-                    int_model.data[:, start_row: stop_row, :] = resultslice[5]
-                    int_model.dq[:, start_row: stop_row, :] = resultslice[6]
-                    int_model.var_poisson[:, start_row: stop_row, :] = resultslice[7]
-                    int_model.var_rnoise[:, start_row: stop_row, :] = resultslice[8]
-                    int_model.err[:, start_row: stop_row, :] = resultslice[9]
-                if resultslice[11] is not None:  # Optional Results exist
-                    opt_model.slope[:, :, start_row: (k + 1) * rows_per_slice, :] = resultslice[11]
-                    opt_model.sigslope[:, :, start_row: (k + 1) * rows_per_slice, :] = resultslice[12]
-                    opt_model.var_poisson[:, :, start_row: (k + 1) * rows_per_slice, :] = resultslice[13]
-                    opt_model.var_rnoise[:, :, start_row: (k + 1) * rows_per_slice, :] = resultslice[14]
-                    opt_model.yint[:, :, start_row: (k + 1) * rows_per_slice, :] = resultslice[15]
-                    opt_model.sigyint[:, :, start_row: (k + 1) * rows_per_slice, :] = resultslice[16]
-                    opt_model.pedestal[:, start_row: (k + 1) * rows_per_slice, :] = resultslice[17]
-                    opt_model.weights[:, :, start_row: (k + 1) * rows_per_slice, :] = resultslice[18]
-                    opt_model.crmag[:, :, start_row: (k + 1) * rows_per_slice, :] = resultslice[19]
-            k = k + 1
-
-        return out_model, int_model, opt_model
-
-
-def create_output_models(input_model, number_of_integrations, save_opt,
```

Comment:
Has create_output_models( ) been replaced by another function ?

Reply:
No.  There are no models being returned from ramp fitting anymore.  Ramp fitting returns tuples of arrays that the step code in each pipeline will now use to create the desired data model specific to that pipeline.